common/tools/dev-tool/src/config/rollup.base.config.ts

```diff
@@ -83,9 +84,16 @@ function ignoreChaiCircularDependencyWarnings(warning: RollupWarning): boolean {
   );
 }
 
+function ignoreOpenTelemetryThisIsUndefinedWarnings(warning: RollupWarning): boolean {
+  return (
+    warning.code === "THIS_IS_UNDEFINED" && warning.id?.includes("@opentelemetry/api") === true
+  );
+}
+
```

Comment:
is this warning the same one that was handled by `openTelemetryCommonJs`?

Reply:
This is a different one:

```
dist-esm/test/**/*.spec.js, dist-esm/test/**/node/** → dist-test/index.browser.js...
(!) `this` has been rewritten to `undefined`
https://rollupjs.org/guide/en/#error-this-is-undefined
../../../common/temp/node_modules/.pnpm/@opentelemetry+api@1.0.3/node_modules/@opentelemetry/api/build/esm/api/context.js
14:  * limitations under the License.
15:  */
16: var __spreadArray = (this && this.__spreadArray) || function (to, from) {
                         ^
17:     for (var i = 0, il = from.length, j = to.length; i < il; i++, j++)
18:         to[j] = from[i];
...and 1 other occurrence
../../../common/temp/node_modules/.pnpm/@opentelemetry+api@1.0.3/node_modules/@opentelemetry/api/build/esm/context/NoopContextManager.js
14:  * limitations under the License.
15:  */
16: var __spreadArray = (this && this.__spreadArray) || function (to, from) {
                         ^
17:     for (var i = 0, il = from.length, j = to.length; i < il; i++, j++)
18:         to[j] = from[i];
...and 1 other occurrence
```

In this case `this` is undefined is fine, since the tslib helper that's emitted (At least that's what I think is happening here - correct me if I'm wrong) is valid JS and won't cause runtime issues. 

I wonder why it's being emitted from OTel - but given we are going to be removing OTel as a dependency of our SDK I don't know if it's worth investigating further - what do you think?

Reply:
I think your synopsis is correct and that we can suppress this warning. It's a warning because sometimes it's a problem and sometimes it's not. In this case it's not because top-level `this` is just part of the polyfill of array spreading in tslib.

It must choose to emit this helper because the OTel compile target is not ES6 or higher. This is a little troubling because it likely means that we are not picking up the `module` field of the OTel API package. Their CommonJS files are ES5, but ESMs are ES6.